x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/Pivot.java

```diff
@@ -82,15 +83,15 @@ public void validateQuery(Client client, SourceConfig sourceConfig, final Action
 
         client.execute(SearchAction.INSTANCE, searchRequest, ActionListener.wrap(response -> {
             if (response == null) {
-                listener.onFailure(new RuntimeException("Unexpected null response from test query"));
+                listener.onFailure(new ElasticsearchException("Unexpected null response from test query"));
                 return;
             }
             if (response.status() != RestStatus.OK) {
-                listener.onFailure(new RuntimeException("Unexpected status from response of test query: "+ response.status()));
+                listener.onFailure(new ElasticsearchException("Unexpected status from response of test query: "+ response.status()));
                 return;
             }
             listener.onResponse(true);
-        }, e -> listener.onFailure(new RuntimeException("Failed to test query", e))));
+        }, e -> listener.onFailure(new ElasticsearchException("Failed to test query", e))));
```

Comment:
does not look like a 400 to me, but another 5xx one. ElasticsearchStatusException with 503?

All errors here are corner cases: any invalid query would not make it here as we parse the query before this is called. So an invalid query syntax is not catched here.